base/arrayshow.jl

```diff
@@ -268,12 +272,15 @@ end
 
 # typeinfo agnostic
 # n-dimensional arrays
-function show_nd(io::IO, a::AbstractArray, print_matrix::Function, label_slices::Bool)
+show_nd(io::IO, a::AbstractArray, print_matrix::Function, label_slices::Bool) =
+    invokelatest(_show_nd, io, a, print_matrix, label_slices, map(unitrange, axes(a)))
```

Comment:
This stood out to me as a little odd. Are there other places where we default to `invokelatest` like this?

Reply:
It is a little odd. We use a similar trick (with `invoke`) in `CoreLogging`: https://github.com/JuliaLang/julia/blob/6c1824d99769fa79f69eb866ce76df674fd37878/base/logging.jl#L74-L91

because the logging framework is *so* untyped, and also would otherwise be a dependency of a huge number of methods (anything with a `@warn`, `@info`, etc). Since that includes the package-loading code, it seemed urgent to put in a barrier.

This is a much less clear-cut case. See #37741 for the background. I'd be willing to get rid of the `invokelatest` and wrap `a` in `inferencebarrier`. Especially with the axes now inferrable in this PR, that may be pretty safe. Thoughts?

Reply:
I've removed the objectionable `invokelatest`, see if this passes muster. We have a little while to live with this and see whether it's a juicy invalidation target without it.